test/hotspot/jtreg/compiler/codecache/stress/OverloadCompileQueueTest.java

```diff
@@ -52,8 +53,33 @@
 import java.util.stream.IntStream;
 import java.util.Random;
 
-public class OverloadCompileQueueTest implements Runnable {
+class LockUnlockThread extends Thread {
     private static final int MAX_SLEEP = 10000;
+    private static final int DELAY_BETWEEN_LOCKS = 100;
+    private final Random rng = Utils.getRandomInstance();
+
+    public volatile boolean isActive = true;
+
+    @Override
+    public void run() {
+        try {
+            while (isActive) {
+                int timeInLockedState = rng.nextInt(MAX_SLEEP);
+                Helper.WHITE_BOX.lockCompilation();
+                Thread.sleep(timeInLockedState);
+                Helper.WHITE_BOX.unlockCompilation();
+                Thread.sleep(DELAY_BETWEEN_LOCKS);
+            }
+        } catch (InterruptedException e) {
+            throw new Error("TESTBUG: lockUnlocker thread was unexpectedly interrupted", e);
+        } finally {
+            Helper.WHITE_BOX.unlockCompilation();
+        }
+    }
+}
+
+public class OverloadCompileQueueTest implements Runnable {
+    private static final LockUnlockThread lockUnlockThread = new LockUnlockThread();
```

Comment:
`static final` field should be `LOCK_UNLOCK_THREAD`?

Reply:
Fixed by moving the static field into the 'main' method (thus making it function-local)